From e639f7fe410a5780efae4d39449f1bdfa22fdfb4 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:41 -0400
Subject: [fs] btrfs: fix possible memory leak in btrfs_create_tree()

Message-id: <1407179109-13714-335-git-send-email-sandeen@redhat.com>
Patchwork-id: 87323
O-Subject: [RHEL7.1 PATCH 334/422] Btrfs: fix possible memory leak in btrfs_create_tree()
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Tsutomu Itoh <t-itoh@jp.fujitsu.com>

commit 59885b39303edc0b3284f324a1ca42cb6e7742d0
Author: Tsutomu Itoh <t-itoh@jp.fujitsu.com>
Date:   Wed Apr 9 09:18:04 2014 +0900

    Btrfs: fix possible memory leak in btrfs_create_tree()

    In btrfs_create_tree(), if btrfs_insert_root() fails, we should
    free root->commit_root.

    Reported-by: Alex Lyakas <alex@zadarastorage.com>
    Signed-off-by: Tsutomu Itoh <t-itoh@jp.fujitsu.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/disk-io.c b/fs/btrfs/disk-io.c
index c92910d..20340d1 100644
--- a/fs/btrfs/disk-io.c
+++ b/fs/btrfs/disk-io.c
@@ -1368,6 +1368,7 @@ struct btrfs_root *btrfs_create_tree(struct btrfs_trans_handle *trans,
 fail:
  if (leaf) {
   btrfs_tree_unlock(leaf);
+  free_extent_buffer(root->commit_root);
   free_extent_buffer(leaf);
  }
  kfree(root);
-- 
1.7.1